pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -439,27 +440,55 @@ public String toString(Object value) {
 
   SHORT_ARRAY,
 
+  PRIMITIVE_INT_ARRAY {
```

Comment:
Do we need to have both INTEGER_ARRAY and PRIMITIVE_INT_ARRAY ?
We can just use the latter right?

Secondly, INT or INTEGER is a primitive type (at least logically). None of them is a complex type. But looks like we are differentiating between both for multi-value columns? In that case what about INT (PRIMITIVE_INT) v/s INTEGER for single valued columns ?

Reply:
`INTEGER_ARRAY` represents `Integer[]` while `PRIMITIVE_INT_ARRAY` represents `int[]`. The reason why we have to treat MV differently is because `Integer[]` and `int[]` cannot be auto casted in java.

Reply:
Got it. Discussed offline that in future we might want to refactor the ingestion code to get rid of Object[] arrays if possible. This will allow us to remove INTEGER_ARRAY and only keep PRIMITIVE_INT_ARRAY for both ingestion and execution